packages/h3/src/H3Adapter.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  AppControllerRoute,
+  AppViewRoute,
+  BullBoardQueues,
+  ControllerHandlerReturnType,
+  IServerAdapter,
+  UIConfig,
+} from '@bull-board/api/dist/typings/app';
+import { readFileSync, statSync } from 'fs';
+import { resolve, normalize } from 'node:path';
+import {
+  createRouter,
+  eventHandler,
+  getRouterParams,
+  getQuery,
+  serveStatic,
+  createError,
+} from 'h3';
+import ejs from 'ejs';
+import { getContentType } from './utils/getContentType';
+import { HTTPMethod } from '@bull-board/api/dist/typings/app';
+
+export class H3Adapter implements IServerAdapter {
+  private uiHandler = createRouter();
+  private basePath = '';
+  private entryRoute: AppViewRoute | undefined;
+  private statics: { path: string; route: string } | undefined;
+  private errorHandler: ((error: Error) => ControllerHandlerReturnType) | undefined;
+  private bullBoardQueues: BullBoardQueues | undefined;
+  private viewPath: string | undefined;
+  private uiConfig: UIConfig = {};
+
+  public setBasePath(path: string): H3Adapter {
+    this.basePath = path;
+    return this;
+  }
+
+  public setStaticPath(staticsRoute: string, staticsPath: string): H3Adapter {
+    this.statics = { route: staticsRoute, path: staticsPath };
+
+    return this;
+  }
+
+  public setViewsPath(viewPath: string): H3Adapter {
+    this.viewPath = viewPath;
+
+    return this;
+  }
+
+  public setErrorHandler(handler: (error: Error) => ControllerHandlerReturnType) {
+    this.errorHandler = handler;
+
+    return this;
+  }
+
+  public setApiRoutes(routes: AppControllerRoute[]): H3Adapter {
+    routes.forEach(({ route, handler, method: methodOrMethods }) => {
+      const methods = Array.isArray(methodOrMethods) ? methodOrMethods : [methodOrMethods];
+
+      methods.forEach((method) => {
```

Comment:
redundant loop, `router.use` of H3 supports an array of methods, so you can pass the `methods` to `this.registerRoute`.